lib/kube/kubeconfig/localproxy.go

```diff
@@ -67,10 +66,10 @@ type LocalProxyValues struct {
 	TeleportKubeClusterAddr string
 	// LocalProxyURL is the local forward proxy's URL.
 	LocalProxyURL string
-	// LocalProxyCAPaths are the paths to local proxy's self-signed CA by Teleport cluster name.
-	LocalProxyCAPaths map[string]string
-	// ClientKeyPath is the path to the client key.
-	ClientKeyPath string
+	// LocalProxyCAs are the local proxy's self-signed CAs PEM encoded data, by Teleport cluster name.
```

Comment:
@greedy52 I've changed it to be the data itself, instead of relying on paths because it's easier to keep it like this now, when relogin is supported. After relogin user's key is changed and CAs and ephemeral kubeconfig were deleted. We now generate our own key and CA for the local proxy, not related to the user's key, so listener is not affected after relogin. And it's simpler to restore just one file of ephemeral kubeconfig with everything inside it.

Reply:
interesting. Are we trying to preserve `tsh proxy kube` across user relogin?

I could be wrong but i believe not only the ephemeral kubeconfig gets removed, default config is also reverted for any `tsh kube login`?

I think storing data instead of paths it's a good idea. Do you think if we could still load from user's key (and the CA generated from it) when `tsh proxy kube` starts? Just to avoid re-generating the local CAs every time. I think regenerating for the long-live `tsh proxy kube` session is not a big deal. But for short-lived local proxy sessions (e.g i'm adding auto-start-local-proxy for `tsh kubectl/kube exec`), it would be great to re-use the same ones if it's there.

Reply:
Yep, trying to preserve proxy across user relogin, same as DB proxy tunnel does.
We remember our ephemeral kubeconfig on start of proxy and then default config changes don't affect it. And proxy will reissue certificates on demand.
I see your point about not re-generating local CA every time for short-live sessions 🤔 I'll look into it.

Reply:
do we rewrite the ephemeral kubeconfig for clients to use?

Reply:
Yep, we rewrite it on relogin https://github.com/gravitational/teleport/pull/24527/files#diff-48fa678c248cdfc9807750af3e73719ef0289f5d18102854f3abba35bab35c37R473-R480